### PR TITLE
Add checksum for libisl 0.24 and update URL

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed new (and existing) sha256.  Will defer final decision to Adam since he contributed the majority of the package.

**Comment by adamjstewart** [comment]
I don't know anything about this library other than that it's a GCC dependency. I've contributed to just about every package in Spack at some point or another. These days I'm really only maintaining packages where I'm listed as a maintainer (Python + geospatial libs).

**Comment by tldahlgren** [comment]
> I don't know anything about this library other than that it's a GCC dependency. I've contributed to just about every package in Spack at some point or another. These days I'm really only maintaining packages where I'm listed as a maintainer (Python + geospatial libs).

👍 I will *try* to keep that in mind.

**Pull request merged**
